SofaKernel/modules/SofaCore/src/sofa/core/loader/MeshLoader.cpp

```diff
@@ -832,30 +831,42 @@ void MeshLoader::copyMeshToData(sofa::helper::io::Mesh* _mesh)
 {
     // copy vertices
     helper::vector<sofa::defaulttype::Vec3>& my_positions = *(d_positions.beginEdit());
-    my_positions = _mesh->getVertices();
+    helper::vector<sofa::defaulttype::Vec3> allVertices = _mesh->getVertices();
```

Comment:
You are doing a whole copy here so use a (const) reference instead
`const helper::vector<sofa::defaulttype::Vec3>& allVertices = _mesh->getVertices();`
(same for edges, triangles, etc)

Reply:
More for my knowledge than a change request, instead of:
```
helper::vector<sofa::defaulttype::Vec3>& my_positions = *(d_positions.beginEdit());
    helper::vector<sofa::defaulttype::Vec3> allVertices = _mesh->getVertices();
    for(size_t i=0; i<allVertices.size(); i++)
        this->addPosition(&my_positions, allVertices[i]);
    d_positions.endEdit();
```

would that work?
```
d_positions.setValue(_mesh->getVertices());
```

Reply:
ah sorry... too early review.. I read a simple copy, not calling a method. Forget my comment.